### PR TITLE
WIP - Provide a setting to flush logs every n bytes.

### PR DESCRIPTION
**Pull request by garethson**
#### Tl;DR; This PR allows us to split very large buffered logging entries with hundreds of lines into separate entries with sane sizes, along with the contextual information required for each entry.

One issue with using this buffered logger is that your logging statements can be arbitrarily split up or even dropped in downstream systems based on certain conditions (byte size, # of lines, etc).

If one has contextual identifiable data at the head of each buffered logger entry (request or job params), it means the subsequent split up lines that aren't dropped altogether are effectively lost, as one usually cannot search for them properly.

This PR adds the ability to specify a maximum byte size for each log entry. When entries are flushed, we provide pre and post flush hooks to re-add any information one wants to make the subsequent log  entries identifiable.

Example usage:

```ruby
# frozen_string_literal: true

BufferedLogger.configure do |config|
  config.flush_at_byte_size = 1000
  config.before_flush = proc { Rails.logger.info("Flushing logs at 1000 bytes") }
  config.after_flush = proc { Rails.logger.info("Request_id: #{RequestContext.request_id}") }
end

```

Note that the `flush_at_byte_size` is not a hard limit, in  that we flush *after* appending the last entry that crosses this threshold, in order to not cut off data. This value should be well under the maximum allowed log entry length of your downstream log parsing.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -4,6 +4,5 @@ rvm:
   - 2.1.0
   - 2.0.0
 env:
-  - "AS_VERSION=3.2.16"
-  - "AS_VERSION=4.0.2"
+  - "AS_VERSION=4.2"
```
Comment (garethson) [inline .travis.yml]: This was causing issues with our minitest version, imo we don't need to support this old stuff.

Comment (kmcphillips) [reply .travis.yml]: For sure. I think even 5.2 would be fine.
